core/cache/unit.go

```diff
@@ -73,6 +73,10 @@ func (u *Unit) Ports() []network.Port {
 // Config settings returns the effective charm configuration for this unit
 // taking into account whether it is tracking a model branch.
 func (u *Unit) ConfigSettings() (charm.Settings, error) {
+	if u.details.CharmURL == "" {
+		return nil, errors.New("unit charm not set")
+	}
```

Comment:
Is this indicating that we didn't get an event? Is this just a debugging prophylactic?

Reply:
This makes it congruent with behaviour of the state-based watcher. Existing unit tests assert this behaviour.